pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -356,6 +366,13 @@ func (s *Service) getOrCreateMonitor(openStackCluster *infrav1.OpenStackCluster,
 		Timeout:        5,
 	}
 	monitor, err = s.loadbalancerClient.CreateMonitor(monitorCreateOpts)
+	// Skip creating monitor if it is not supported by Octavia provider
+	if capoerrors.IsNotImplementedError(err) {
+		record.Warnf(openStackCluster, "SkippedCreateMonitor", "Health monitors are not supported with the current Octavia provider.")
+		record.Eventf(openStackCluster, "SkippedCreateMonitor", "Health Monitor is not created as it's not implemented with the current provider.")
```

Comment:
This message should be updated in a similar fashion as it's user-facing. I added "Octavia" on purpose, to avoid confusion what "provider" means here. An inexperienced user may think that we're saying cluster-api-*provider*-openstack doesn't support health monitors.